smack-extensions/src/main/java/org/jivesoftware/smackx/message/correct/MessageCorrectExtension.java

```diff
@@ -0,0 +1,120 @@
+/**
+ *
+ * Copyright the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.message.correct;
+
+import java.io.IOException;
+
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.provider.ExtensionElementProvider;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ * An Extension that implements XEP-0308: Last Message Correction
+ * 
+ * This extension is expected to be added to message stanzas. Please refer to
+ * the XEP for more implementation guidelines.
+ * 
+ * @author Fernando Ramirez, f.e.ramirez94@gmail.com
+ */
+public class MessageCorrectExtension implements ExtensionElement {
```

Comment:
It's Smackomatic (a Smack Idiom) that ExtensionElements have a static `from()` method. Since XEP-308 is only defined for messages, in this case we should have

`static MessageCorrectionExtension from(Message)`

Please add this method.